oauth2_provider/views.py

```diff
@@ -58,10 +63,43 @@ class AccessTokenView(provider.oauth2.views.AccessTokenView):
     and available to the claim handlers configured by `OAUTH_OIDC_ID_TOKEN_HANDLERS`
 
     """
+    grant_types = provider.oauth2.views.AccessTokenView.grant_types + ["edx_session"]
 
     # Add custom authentication provider, to support email as username.
     authentication = (provider.oauth2.views.AccessTokenView.authentication +
-                      (PublicPasswordBackend, ))
+                      (PublicPasswordBackend, ExistingSessionBackend))
+
+    def edx_session(self, request, data, client):
+        """
+        Handle ``grant_type=edx_session`` requests.
+        """
+        grant_data = self.get_edx_session_grant(request, request.POST, client)
+        at = self.create_access_token(request, grant_data['user'], grant_data['scope'], client)
```

Comment:
For anyone as confused as I was, I forgot that we use our own fork of django-oauth2-provider, whose implementation of [AccessToken](https://github.com/edx/django-oauth2-provider/blob/edx/provider/views.py#L346) includes [`access_token_response_data()`](https://github.com/edx/django-oauth2-provider/blob/edx/provider/views.py#L468-L490), a method not present on django-oauth2-provider's master branch.